src/Bicep.Core/Syntax/ImportSpecification.cs

```diff
@@ -46,35 +49,57 @@ private ImportSpecification(string name, string version, bool isValid, TextSpan
-
-                    return new ImportSpecification(name, version, isValid, span);
-
-                case SkippedTriviaSyntax trivia:
-                    return new ImportSpecification(trivia.TriviaName, trivia.TriviaName, false, trivia.Span);
+                StringSyntax stringSyntax when stringSyntax.TryGetLiteralValue() is { } value
+                    => CreateFromStringSyntax(stringSyntax, value),
+                SkippedTriviaSyntax trivia
+                    => new ImportSpecification(
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        trivia.TriviaName,
+                        false,
+                        trivia.Span),
+                _
+                    => new ImportSpecification(
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        LanguageConstants.ErrorName,
+                        false,
+                        specificationSyntax.Span)
+            };
+        }
 
-                default:
-                    return new ImportSpecification(LanguageConstants.ErrorName, LanguageConstants.ErrorName, false, specificationSyntax.Span);
+        public SyntaxBase ToPath()
+        {
+            if (!this.IsValid)
+            {
+                return new SkippedTriviaSyntax(this.Span, Enumerable.Empty<SyntaxBase>());
             }
+            return SyntaxFactory.CreateStringLiteral(this.UnqualifiedAddress);
         }
 
-        private static (string Name, string Version, bool IsValid) Parse(string value)
+        private static ImportSpecification CreateFromStringSyntax(StringSyntax stringSyntax, string value)
         {
-            var match = SpecificationPattern.Match(value);
-
-            if (!match.Success)
+            var matchBareSpecification = BareSpecification.Match(value);
+            if (!matchBareSpecification.Success)
             {
-                return (LanguageConstants.ErrorName, LanguageConstants.ErrorName, false);
+                return new(
+                 LanguageConstants.ErrorName,
+                 LanguageConstants.ErrorName,
+                 LanguageConstants.ErrorName,
+                 false,
+                 stringSyntax.Span);
             }
 
-            var name = match.Groups["name"].Value;
-            var version = match.Groups["version"].Value;
+            var name = matchBareSpecification.Groups["name"].Value;
+            var version = matchBareSpecification.Groups["version"].Value;
+            var parts = value.Split('@');
+            var artifactAddress = parts[0];
+            
+            var span = new TextSpan(stringSyntax.Span.Position + 1, artifactAddress.Length);
 
-            return (name, version, true);
+            return new(string.Join(':', parts), name, version, artifactAddress != "az", span);
```

Comment:
This logic isn't very clear:
* Why are you doing `value.Split()` instead of using the regex capture groups?
* What does `parts` represent, and why are you joining it with `:` on line 102?
* Why the rename from `SpecificationPattern` -> `BareSpecification`? What does "bare" mean in this context?
* Why is the snytax considered "invalid" if `artifactAddress != "az"`?

Reply:
> Why are you doing value.Split() instead of using the regex capture groups?

The capture groups are not capturing the full `TriviaAddress`, since I removed the `'^'` char, `Name` only captures the implicit symbol defined in the specification.
For the processing in `ModuleDispatcher` we need the `TriviaAddress` in a format that is aligned with module specification. So we need to replace the `@` character with `:`.

> What does parts represent, and why are you joining it with : on line 102?

the name `parts` is used in this context to call out the two parts that conform the specification, the left part which is the repository trivia address (left of the `@` sign) and the right part (right of the `@` sign) which is the version descriptor.  

> Why the rename from SpecificationPattern -> BareSpecification? What does "bare" mean in this context?

Bare in the context of this processing means "only the implicit symbol". But I am happy to rename if you have a suggestion. Naming is hard 👍 
 
> Why is the snytax considered "invalid" if artifactAddress != "az"?

I need a way to distinguish a bare specification like `import 'kubernetes@1.0.0'` or `import 'sys@1.0.0'` which continue to be valid from `import 'az@1.0.0.'` which is no longer a valid specification per the approved design https://github.com/Azure/bicep/discussions/12110

Reply:
Some thoughts on how to possibly structure the code differently to make some of these things clearer:

1. Really we are handling two distinct patterns with different rules - "built-in" and "from-registry". Handling both in one place is terse, but is also difficult to follow. You could differentiate the two in the code e.g.:
    ```csharp
    if (BuiltInSpecificationPattern.Match(...) is {} builtInMatch &&
        builtInMatch.Success)
    {
        // ... old logic here (with a comment explaining what 'built-in' means)
    }
    
    if (FromRegistrySpecificationPattern.Match(...) is {} registryMatch &&
        registryMatch.Success)
    {
        // ... new logic to parse out the "name" from the spec path goes here (again with a comment)
    }
    
    // ... error handling...
    ```
1. `Bare` doesn't have a clear meaning to me (hopefully my previous point addresses this)
1. If we're going to the trouble of writing a regex, it feels odd to then mix this with use of `Split()` - could you instead define a capture group for the `unqualifiedAddress`?
1. Does it make sense to set `unqualifiedAddress` for a 'built-in" pattern? Could we instead make this nullable and just set it to `null`?
1. > import 'az@1.0.0.' which is no longer a valid specification per the approved design
    
    - What's the benefit of enforcing this here? It doesn't feel like an appropriate concern for this part of the code - for example, we're not blocking `'aws@1.0.0'` or `'imnotarealmodule@1.0.0'`.
    - What's the reason for blocking it generally? Does permitting it create any problems?